### ISSUE TITLE
"unresolved import" when using "paste! {...}"

### ISSUE THREAD

**Issue by kehlert** [rust-analyzer/rust-analyzer#6089]
I am using the [paste crate](https://docs.rs/paste/1.0.1/paste/), and and rust-analyzer reports an error in VSCode for the below example code. However, the code compiles and `cargo check` does not report any errors.

Here is the error
<img width="475" alt="Screen Shot 2020-09-28 at 10 28 37 PM" src="https://user-images.githubusercontent.com/2395933/94505855-ff701a80-01d9-11eb-9c31-caf2b9c663da.png">

And here is the example code

_main.rs_
```rust
mod my_struct;

use my_struct::MyStruct;

fn main() {
    println!("Hello, world!");
}
```

_my_struct.rs_
```rust
use paste::paste;

paste! {
    pub struct MyStruct;
}
```

Even if I put the code all into one file, there are still issues with resolving the type. Is there any workaround? Can I at least suppress the errors?

**Comment by bjorn3** [comment]
You need to enable proc-macro support using

```jsonc
{
    "rust-analyzer.cargo.loadOutDirsFromCheck": true,
    "rust-analyzer.procMacro.enable": true,
}
```
